### PR TITLE
Separate game engine from user interface

### PR DESCRIPTION
**Pull request by RauliL**
Split game engine and user interface into two separate files. Game engine API will be exported as global variable called `GameEngine` which the user interface then interacts with. Move the current user interface into directory `theme-default`.

### CODE REVIEW

File: engine.js [hunk 1]
```diff
@@ -44,7 +52,6 @@ $(() => {
 
   function clearBlock(block) {
     block.color = '';
```
Comment (frostburn) [inline engine.js]: This should set color to `null`.

**Comment by frostburn** [comment]
The UI is missing information about the current chain number and time step.

**Comment by RauliL** [comment]
Fixed the issues.

**Comment by frostburn** [comment]
Looks good now. :+1:

**Pull request merged**